app/models/chargeback_rate.rb

```diff
@@ -11,6 +20,8 @@ class ChargebackRate < ApplicationRecord
   validates_uniqueness_of   :guid
   validates_uniqueness_of   :description, :scope => :rate_type
 
+  virtual_column :assigned?, :type => :boolean
```

Comment:
@tenderlove how in your example would the UI ask the model whether this thing can be destroyed so it can grey out the Delete button in the UI and put a hover-over of the reason why it is grey.  At the same time, keep before_destroy working as usual for the actual destroy validation.